test/els_execute_command_SUITE.erl

```diff
@@ -0,0 +1,100 @@
+-module(els_execute_command_SUITE).
+
+%% CT Callbacks
+-export([ suite/0
+        , init_per_suite/1
+        , end_per_suite/1
+        , init_per_testcase/2
+        , end_per_testcase/2
+        , groups/0
+        , all/0
+        ]).
+
+%% Test cases
+-export([ erlang_ls_info/1
+        , strip_server_prefix/1
+        ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include_lib("common_test/include/ct.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+%%==============================================================================
+%% Types
+%%==============================================================================
+-type config() :: [{atom(), any()}].
+
+%%==============================================================================
+%% CT Callbacks
+%%==============================================================================
+-spec suite() -> [tuple()].
+suite() ->
+  [{timetrap, {seconds, 30}}].
+
+-spec all() -> [{group, atom()}].
+all() ->
+  [{group, tcp}, {group, stdio}].
+
+-spec groups() -> [atom()].
+groups() ->
+  els_test_utils:groups(?MODULE).
+
+-spec init_per_suite(config()) -> config().
+init_per_suite(Config) ->
+  els_test_utils:init_per_suite(Config).
+
+-spec end_per_suite(config()) -> ok.
+end_per_suite(Config) ->
+  els_test_utils:end_per_suite(Config).
+
+-spec init_per_testcase(atom(), config()) -> config().
+init_per_testcase(TestCase, Config) ->
+  els_test_utils:init_per_testcase(TestCase, Config).
+
+-spec end_per_testcase(atom(), config()) -> ok.
+end_per_testcase(TestCase, Config) ->
+  els_test_utils:end_per_testcase(TestCase, Config).
+
+%%==============================================================================
+%% Testcases
+%%==============================================================================
+
+-spec erlang_ls_info(config()) -> ok.
+erlang_ls_info(Config) ->
+  Uri = ?config(code_navigation_uri, Config),
+  PrefixedCommand
+    = els_execute_command_provider:add_server_prefix(<<"info">>),
+  #{result := Result}
+    = els_client:workspace_executecommand(PrefixedCommand, [#{uri => Uri}]),
+  Expected = [],
+  ?assertEqual(Expected, Result),
+  timer:sleep(10), %% otherwise the stdio client does not get the Notification
```

Comment:
I have done this myself sometimes, but I think we should use a similar approach to the one in `els_test_utils:wait_for/2` but instead of waiting for a message we can provide a function. This is mainly to avoid flakiness in the tests, which can happen if we run the builds in slower machines.

The approach I'm reffering to would be definind an `els_test_utils:wait_for_fun/3` like so:

```erlang
-spec wait_for_fun(fun(), non_neg_integer(), non_neg_integer()) -> {ok, any()} | timeout.
wait_for_fun(_CheckFun, _WaitTime, 0) ->
  timeout;
wait_for_fun(CheckFun, WaitTime, Retries) ->
  case CheckFun() of
    {true, Value} ->
      {ok, Value};
    false ->
      timer:sleep(WaitTime),
      wait_for_fun(CheckFun, WaitTime, Retries - 1)
  end.
```